pkg/ccl/sqlproxyccl/forwarder.go

```diff
@@ -49,6 +55,32 @@ type forwarder struct {
 	// be closed.
 	errCh chan error
 
+	// timeSource is the source of the time, and uses timeutil.DefaultTimeSource
+	// by default. This is often replaced in tests.
+	timeSource timeutil.TimeSource
+
+	// state represents the state of the forwarder: active or idle. The
+	// forwarder is moved to the idle state by the idle detector if no messages
+	// have been transmitted from/to the client since the last idle check. The
+	// forwarder starts with an active state upon connection establishment.
+	//
+	// WARNING: When acquiring locks on state and mu, they should be acquired in
+	// the following order: state->mu to avoid any potential deadlocks.
+	state struct {
+		syncutil.Mutex
+
+		// lastRequestTransferredAt and lastResponseTransferredAt represent the
+		// logical clock's values that a message was received from / sent to
+		// the client *during the last idle check*. In other words, these
+		// represent snapshot values, and does not get updated in real-time,
+		// unlike the fields in the processors.
+		lastRequestTransferredAt  uint64
```

Comment:
Can we merge this structure with mu? I don't see an immediate benefit to having two separate structures to lock and adding a second mutex increases the risk of deadlocks.